cmd/app/app.go

```diff
@@ -158,6 +161,31 @@ func (o *Options) addFlags(cmd *cobra.Command) {
 	cmd.PersistentFlags().StringVar(&o.Client.Docker.LoginURL,
 		"docker-login-url", "https://hub.docker.com/v2/users/login/",
 		"URL to login into docker using username/password.")
+	cmd.PersistentFlags().StringVar(&o.Client.Selfhosted.Username,
+		"selfhosted-username", "",
+		fmt.Sprintf(
+			"Username is authenticate with a selfhosted registry (%s_%s).",
+			envPrefix, envSelfhostedUsername,
+		))
+	cmd.PersistentFlags().StringVar(&o.Client.Selfhosted.Password,
+		"selfhosted-password", "",
+		fmt.Sprintf(
+			"Password is authenticate with a selfhosted registry (%s_%s).",
+			envPrefix, envSelfhostedPassword,
+		))
+	cmd.PersistentFlags().StringVar(&o.Client.Selfhosted.Bearer,
+		"selfhosted-token", "",
+		fmt.Sprintf(
+			"Token is authenticate with a selfhosted registry. Cannot be used with "+
+				"username/password (%s_%s).",
+			envPrefix, envSelfhostedBearer,
+		))
+	cmd.PersistentFlags().StringVar(&o.Client.Selfhosted.LoginURL,
+		"selfhosted-login-url", "",
+		"URL to login into selfhosted registry using username/password.")
+	cmd.PersistentFlags().StringVar(&o.Client.Selfhosted.URL,
```

Comment:
Do we want to just require the host here?

I assume all registries use the `/v2/` root path. Are there scenarios of this not being the root path?

The docker v2 API spec suggests this should always be the root.

Reply:
- I don't think it should be required as one could be free in what type of "client" they use. If we force this URL, then a person who only uses quay would be required to fill this in?
- I could imagine situations where this client could work for other applications. So if we leave this as configuration level, there are no hardcoded paths.  

In fairness, I'm not to sure about this part either ;)

Reply:
That makes sense. Let's keep it as is, since we're both not sure and this is the safest option.